### PR TITLE
refac: fix charge_type

### PR DESCRIPTION
**Pull request by johevemi**

### CODE REVIEW

**Review by energinet-ajw — APPROVED** [review]

File: source/databricks/calculation_engine/package/calculation/preparation/transformations/charge_types.py [hunk 1]
```diff
@@ -138,19 +138,22 @@ def _join_with_metering_points(df: DataFrame, metering_points: DataFrame) -> Dat
...
+                _get_window_duration_string_based_on_resolution(charge_resolution),
+            ),
         ],
-    ).select(
+    ).selectExpr(
         Colname.sum_quantity,
         Colname.qualities,
         Colname.metering_point_id,
-        Colname.observation_time,
+        f"window.{Colname.start} as {Colname.observation_time}",
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/package/calculation/preparation/transformations/charge_types.py:156]: How about aliasing the window column and thus preventing resorting to using `selectExpr` and strings?

File: source/databricks/calculation_engine/package/calculation/preparation/transformations/charge_types.py [hunk 2]
```diff
@@ -166,6 +169,20 @@ def _group_by_time_series_on_metering_point_id_and_resolution_and_sum_quantity(
     return grouped_time_series
 
 
+def _get_window_duration_string_based_on_resolution(
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/package/calculation/preparation/transformations/charge_types.py:172]: Seems like something we'd very likely be using elsewhere later. Is it a general functionality that ought to be located somewhere easier to find and resuse?

File: source/databricks/calculation_engine/tests/calculation/preparation/transformations/test_charge_types.py [hunk 3]
```diff
@@ -539,3 +539,138 @@ def test__get_tariff_charges__returns_df_with_expected_values(
 
     # Assert
     assert actual.collect() == expected_tariff_charges.collect()
+
+
+@pytest.mark.parametrize(
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/tests/calculation/preparation/transformations/test_charge_types.py:544]: The test file is already big and now it has become significantly larger.
How about splitting the file by what function is being tested? This also caters to shorter test names.

File: source/databricks/calculation_engine/package/calculation/preparation/transformations/charge_types.py (hunk 1 above)
Comment (johevemi) [reply source/databricks/calculation_engine/package/calculation/preparation/transformations/charge_types.py]: i will have to write col("window.start")

File: source/databricks/calculation_engine/package/calculation/preparation/transformations/charge_types.py (hunk 2 above)
Comment (johevemi) [reply source/databricks/calculation_engine/package/calculation/preparation/transformations/charge_types.py:183]: if we use it later we can move it then. Unless it is something you really want right now

File: source/databricks/calculation_engine/tests/calculation/preparation/transformations/test_charge_types.py (hunk 3 above)
Comment (johevemi) [reply source/databricks/calculation_engine/tests/calculation/preparation/transformations/test_charge_types.py:544]: we can split it into fee, subscription and tariff. the tariff one will still be pretty big tho

Comment (BjarkeMeier) [reply source/databricks/calculation_engine/tests/calculation/preparation/transformations/test_charge_types.py:544]: Sounds like a good start.

**Review by energinet-ajw — APPROVED** [review]

**Pull request merged**